### PR TITLE
Documentation and examples for advanced usage

### PR DESCRIPTION
**Pull request by WyriHaximus**
Currently once can use the `Executor` directly to query but this isn't documented. This PR adds documentation and an example.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,6 +93,31 @@ $dns = $factory->createCached('8.8.8.8', $loop, $cache);
 
 See also the wiki for possible [cache implementations](https://github.com/reactphp/react/wiki/Users#cache-implementations).
 
+## Advanced Usage
+
+For more advanced usages one can utilize the `Executor` directly.
```
Comment (clue) [inline README.md]: I'm planning to deprecate the `Executor` in future versions, so perhaps we should link to the whole namespace instead (may keep the below example in place though)?

File: README.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ easily be used to create a DNS server.
 * [Basic usage](#basic-usage)
 * [Caching](#caching)
   * [Custom cache adapter](#custom-cache-adapter)
+* [Advanced usage](#advance-usage)
```
Comment (clue) [inline README.md]: minor typo here, otherwise LGTM :+1:

File: README.md (hunk 1 above)
Comment (kelunik) [inline README.md]: Deprecating executor seems to be a good idea.

**Comment by clue** [comment]
Changes LGTM, unrelated build error has been addressed via #68 in the meantime :shipit:

**Pull request merged**